Packs/CommonScripts/Scripts/GenerateAsBuiltConfiguration/GenerateAsBuiltConfiguration.py

```diff
@@ -0,0 +1,744 @@
+                        if new.get('type') == 'playbook':
+                            test_d.add(new.get('playbookId'))
+                        # if 'brand' in new.keys():
+                        #     int_name.add(new.get('brand'))
+                        if 'scriptId' in new.keys():
+                            task_name.add(new.get('scriptId'))
+
+    command_list = playbook.get("commands")
+    if configuration is not None:
+        for c in command_list:
+            for k, v in configuration.items():
+                if c in v["commands"]:
+                    int_data.append(k)
+    return test_d, task_name, int_data
+
+
+def create_config_file(pb_name, ignore_playbook):
+    global autodata
+    playbook = get_playbook_data(pb_name)
+    filter_play, filter_auto, filter_int = sub_data(playbook)
+    playbook["dependencies"] = get_playbook_dependencies(playbook)
+    # print("Playbook dependencies", playbook["dependencies"])
+    playbook["automation"] = get_playbook_automation(playbook, filter_auto)  # filter_auto
+    playbook["integration"] = get_playbook_integration(playbook, filter_int)
+    playbook["subplaybook"] = get_playbook_subplaybook(playbook, filter_play)
+    del playbook["dependencies"]
+    if playbook["automation"] is not None:
+        autodata = True
+        for k, v in playbook["automation"].items():
+            auto_id = v.get("id")
+            resp = json.dumps(get_api_request(f"/automation/export/{auto_id}"))
+            resp = (resp.split("script: |")[1])
+            resp = (resp.split("type: python")[0])
+            resp = resp.lstrip("\\n ")
+            resp = resp.rstrip("\\n")
+            auto_script[auto_id] = resp
+            # print(auto_id)
+            # print("Dictionary for automation code",auto_script)
+        playbook["scripts"] = auto_script
+    if playbook["subplaybook"] is not None:
+        # playbook["automation"] = {k: v for k, v in playbook["automation"].items() if v.get('id') in ig_list[1]}
+        # playbook["integration"] = {k: v for k, v in playbook["integration"].items() if k in ig_list[2]}
+        # playbook["subplaybook"] = {k: v for k, v in playbook["subplaybook"].items() if v.get('id') in ig_list[0]}
+        for subplaybook_name in playbook["subplaybook"]:
+            if not playbook["subplaybook"][subplaybook_name]["system"]:
+                ignore_playbook.append(subplaybook_name)
+                playbook["subplaybook"][subplaybook_name] = create_config_file(subplaybook_name, ignore_playbook)
+        return dict(playbook)
+    else:
+        return dict(playbook)
+
+
+def create_as_built(playbook_names, ignore_playbook):
+    configuration_data = []
+    for pb_name in playbook_names:
+        playbook = create_config_file(pb_name, ignore_playbook)
+        configuration_data.append(playbook)
+    # asbuilt = json.dumps(configuration_data, indent=4)
+    # print("type",type(asbuilt))
+    return configuration_data
+
+
+def main():  # pragma: no cover
```

Comment:
Please put the code in `main` in a `try-except` block. You can find examples of this throughout the repo.